test/test_nn.py

```diff
@@ -8010,6 +8010,19 @@ def test_RNN_dropout(self):
                     self.assertEqual(hy.data[0][0][0], 10)
                     self.assertEqual(hy.data[1][0][0], output_val)
 
+    def test_error_RNN_seq_len_zero(self):
+        # checking error message when RNN has seq_len = 0
+        for module in (nn.RNN, nn.LSTM, nn.GRU):
+            for bidirectional, cuda in product((True, False), repeat=2):
```

Comment:
The CI has multiple configurations. This tests gets run on configurations with CUDA and configurations without CUDA.

When this test is run on a configuration without CUDA, `rnn.cuda()` raises an RuntimeError because there is no CUDA device.

Now, the way to get around this is to get a list of "available devices":
https://github.com/pytorch/pytorch/pull/60059/files#diff-1810aec643bd72f2dbac7f18cbc97d173a9b3b494c389020d28ef49d8e775b28R300-R301

`torch.testing.get_all_device_types()` returns all available device types. So if we are on a CPU-only configuration, it'll return the CPU device. IF we are on a CUDA configuration, it'll return a CPU device and a CUDA device.

Reply:
this is actually very nice code pointer, thanks for making it very clear.